src/ClientWidgetApi.ts

```diff
@@ -587,46 +587,53 @@ export class ClientWidgetApi extends EventEmitter {
             });
         }
 
-        const result = await this.driver.readEventRelations(
-            request.data.event_id, request.data.room_id, request.data.rel_type,
-            request.data.event_type, request.data.from, request.data.to,
-            request.data.limit, request.data.direction);
-
-        // check if the user is permitted to receive the event in question
-        if (result.originalEvent) {
-            if (result.originalEvent.state_key !== undefined) {
-                if (!this.canReceiveStateEvent(result.originalEvent.type, result.originalEvent.state_key)) {
-                    return this.transport.reply<IWidgetApiErrorResponseData>(request, {
-                        error: { message: "Cannot read state events of this type" },
-                    });
-                }
-            } else {
-                if (!this.canReceiveRoomEvent(result.originalEvent.type, result.originalEvent.content['msgtype'])) {
-                    return this.transport.reply<IWidgetApiErrorResponseData>(request, {
-                        error: { message: "Cannot read room events of this type" },
-                    });
+        try {
+            const result = await this.driver.readEventRelations(
+                request.data.event_id, request.data.room_id, request.data.rel_type,
+                request.data.event_type, request.data.from, request.data.to,
+                request.data.limit, request.data.direction);
```

Comment:
Just a small code style nit:

```suggestion
                request.data.limit, request.data.direction,
            );
```